python/nammu/test/test_nammu.py

```diff
@@ -243,3 +251,115 @@ def test_arabic_split_pane(self, monkeypatch):
+        # edits, but that's a bit overcomplicated at this stage.
+        assert (listener.current_compound != empty_compound and
+                "edits: []" not in controller.undo_manager.toString())
+
+    def test_undo_empty_pane(self):
+        '''
+        Undo when empty pane should not do anything and the undo stackpile
+        should remain empty.
+        This test is needed because undoing empty panels is unstable and
+        sometimes raises exceptions.
+        '''
+        controller = self.nammu.atfAreaController
+        controller.clearAtfArea()
+        controller.undo()
+        assert ("edits: []" in controller.undo_manager.toString())
+
+    def test_redo_empty_pane(self):
+        '''
+        Redo when empty pane should not do anything and the undo stackpile
+        should remain empty.
+        This test is needed because undoing empty panels is unstable and
+        sometimes raises exceptions.
+        '''
+        controller = self.nammu.atfAreaController
+        controller.clearAtfArea()
+        controller.redo()
+        assert ("edits: []" in controller.undo_manager.toString())
+
+    def test_undo_after_opening_file(self, monkeypatch):
+        '''
+        Undo after closing a file should not bring back the old file.
+        '''
+        import javax.swing.JFileChooser
+        monkeypatch.setattr(javax.swing.JFileChooser, 'showDialog',
+                            show_diag_patch)
+        monkeypatch.setattr(javax.swing.JFileChooser, 'getSelectedFile',
+                            selected_file_patch_english)
+        monkeypatch.setattr(self.nammu, 'handleUnsaved', unsaved_patch)
+        self.nammu.openFile()
+        controller = self.nammu.atfAreaController
+        controller.undo()
+        assert ("edits: []" in controller.undo_manager.toString())
+
+    def test_undo_after_closing_file(self, monkeypatch):
+        '''
+        Undo after closing a file should not bring back the old file.
+        '''
+        import javax.swing.JFileChooser
+        monkeypatch.setattr(javax.swing.JFileChooser, 'showDialog',
+                            show_diag_patch)
+        monkeypatch.setattr(javax.swing.JFileChooser, 'getSelectedFile',
+                            selected_file_patch_english)
+        monkeypatch.setattr(self.nammu, 'handleUnsaved', unsaved_patch)
+        self.nammu.openFile()
+        self.nammu.closeFile()
+        controller = self.nammu.atfAreaController
+        controller.undo()
+        assert ("edits: []" in controller.undo_manager.toString())
+
+    def test_undo_edit_pane(self, empty_compound):
+        '''
+        Check adding a simple text and undoing last edit compound works.
+        '''
+        controller = self.nammu.atfAreaController
+        controller.edit_area.setText("Hello Nammu!")
+        controller.undo()
+        listener = controller.view.edit_listener
+        assert (listener.current_compound != empty_compound)
```

Comment:
A summary of what we discussed in person:
It feels like this should be `==` instead of `!=`, but it looks like two empty compounds are not equal according to `==`, nor to `.equals`... Could maybe check `edits: [] in listener.current_compound.toString()` as above?

Reply:
Done.